lib/sensors/include/AEAT6012.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <mutex>
+
+/*
+ * Broadcom's AEAT-6012 is an absolute magnetic encoder with 12 bit resolution and SSI interface
+ * Datasheet: https://docs.broadcom.com/doc/AV02-0188EN
+ */
+
+namespace Encoder {
+typedef void (*callback_ptr)(void);
+
+typedef struct {
+  PinName cs;
+  PinName spi_mosi;
+  PinName spi_clk;
+  float offset_deg;
+} Config;
+
+class AEAT6012 {
+ public:
+  AEAT6012(PinName cs, PinName spi_mosi, PinName spi_clk, float offset_deg = 0.0f);
+  AEAT6012(const Config &config);
+
+  // Trigger a blocking encoder read and return absolute position in degrees
+  bool getAngleDeg(float &angle);
+
+  // Trigger a blocking encoder read and return angular velocity in degrees / second
+  bool getAngularVelocityDegPerSec(float &speed);
+
+  // Reset encoder values
+  bool reset(void);
+
+  // Asynchronous API for triggering encoder read
+  // Invokes user callback once read transaction is complete
+  // Returns true if SPI read successfully started, false if SPI peripheral is busy
+  bool readAsync(callback_ptr callback);
+
+  // Returns stored absolute position in degrees (without invoking an encoder read)
+  float getAngleDegNoTrigger(void);
+
+  // Returns stored raw encoder reading (without invoking an encoder read)
+  uint16_t getPositionRawNoTrigger(void);
+
+  // Returns stored angular velocity in degrees / second (without invoking an encoder read)
+  float getAngularVelocityDegPerSecNoTrigger(void);
```

Comment:
i think the only functions that should be public are whatever ones are defined as virtual in Sensor class. I think sticking to this makes the code simpler and more consistent.

Reply:
What about the async functions (like all the no trigger ones)?

Reply:
Also I realized that the functions that would get called in the async driver's callback don't have the mutex protection cuz mutex acquiring is blocking and can't be in an isr

Reply:
sure you can have the async function public